plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -255,6 +258,19 @@ public void flushCache()
         rolesCache.invalidateAll();
     }
 
+    public void flushPartitionCache(String schemaName, String tableName, List<String> partitionColumns, List<String> partitionValues)
+    {
+        requireNonNull(schemaName, "schemaName is null");
+        requireNonNull(tableName, "tableName is null");
+        requireNonNull(partitionColumns, "partitionColumns is null");
+        requireNonNull(partitionValues, "partitionValues is null");
+
+        String providedPartitionName = makePartName(partitionColumns, partitionValues);
+        Predicate<Optional<String>> partitionPredicate =
+                partitionNameToCheck -> partitionNameToCheck.map(value -> value.equals(providedPartitionName)).orElse(false);
+        invalidatePartitionCache(schemaName, tableName, Optional.of(partitionPredicate));
```

Comment:
fine to inline the predicate:

```java
String partitionNameToInvalidate = makePartName(partitionColumns, partitionValues);
        invalidatePartitionCache(schemaName, tableName, Optional.of(partition -> partition.isPresent() && partition.get().equals(partitionNameToInvalidate)));
```

even more fine when you remove the `Optional`

Reply:
Done